Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -0,0 +1,245 @@
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task</returns>
+        public static Task ExecuteOnUIThreadAsync(Func<Task> function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            return ExecuteOnUIThreadAsync<object>(
+                async () =>
+                {
+                    await function().ConfigureAwait(false);
+                    return null;
+                }, priority);
+        }
+
+        /// <summary>
+        /// Execute the given function asynchronously on given view's UI thread. Default view is the main view.
+        /// </summary>
+        /// <param name="viewToExecuteOn">View for the <paramref name="function"/>  to be executed on </param>
+        /// <param name="function">Asynchronous function to be executed asynchronously on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task/></returns>
+        public static Task ExecuteOnUIThreadAsync(CoreApplicationView viewToExecuteOn, Action function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            if (viewToExecuteOn == null)
+            {
+                throw new ArgumentNullException("viewToExecuteOn can't be null!");
+            }
+
+            return viewToExecuteOn.Dispatcher.AwaitableRunAsync(function, priority);
+        }
+
+        /// <summary>
+        /// Execute the given function asynchronously on given view's UI thread. Default view is the main view.
+        /// </summary>
+        /// <param name="function">Asynchronous function to be executed asynchronously on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task/></returns>
+        public static Task ExecuteOnUIThreadAsync(Action function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            return ExecuteOnUIThreadAsync(CoreApplication.MainView, function, priority);
+        }
+
+        /// <summary>
+        /// Execute the given function asynchronously on given view's UI thread. Default view is the main view.
+        /// </summary>
+        /// <typeparam name="T">returned data type of the function</typeparam>
+        /// <param name="viewToExecuteOn">View for the <paramref name="function"/>  to be executed on </param>
+        /// <param name="function">Synchronous function with return type <typeparamref name="T"/> to be executed on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task with type <typeparamref name="T"/></returns>
+        public static Task<T> ExecuteOnUIThreadAsync<T>(CoreApplicationView viewToExecuteOn, Func<T> function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
```

Comment:
Hmm, that is also a valid point. We can support both, there is no harm as anybody could use any options that they would like and the outcome would be the same. I'll add it in for 1.3 then (since this has already been merged?) cc @deltakosh